src/core/thread/mle.cpp

```diff
@@ -325,6 +325,10 @@ Error Mle::Restore(void)
 
     SuccessOrExit(error = Get<Settings>().Read(networkInfo));
 
+#if OPENTHREAD_MTD
+    VerifyOrExit(!(networkInfo.GetDeviceMode() & DeviceMode::kModeFullThreadDevice));
```

Comment:
should set `error` accordingly?

BTW, what about storing network info but ignore the `D` flag ?

Reply:
> should set error accordingly?

I noticed a few lines below the error is not being set either:
VerifyOrExit(networkInfo.GetVersion() == kThreadVersion);
And anyway the error is ignored here: https://github.com/openthread/openthread/blob/main/src/core/common/instance.cpp#L192

Do you think I should set it anyway?

> BTW, what about storing network info but ignore the D flag ?

If the device was a router before the RLOC16 will be invalid for the new MTD. Same for other info like Child Table. Maybe I should add some logic for forcing re-attach only in the case of Router RLOC16?